.pre-commit-config.yaml

```diff
@@ -46,6 +46,13 @@ repos:
         entry: shfmt -i 2 -ci -sr -kp -w
         types: [shell]
 
+  - repo: https://github.com/golangci/golangci-lint
+    rev: v1.62.2
+    hooks:
+      - id: golangci-lint-full
+        args: [--timeout=5m]
+        language_version: 1.23.4 # renovate: datasource=golang-version
+
```

Comment:
The shell check and the shfmt were next to each because they both focus on bash script, I would put this hook either before or after.